src/main/java/com/aws/greengrass/config/Topic.java

```diff
@@ -53,6 +54,27 @@ public Topic subscribe(Subscriber s) {
         return this;
     }
 
+    /**
+     * Subscribe to a topic and invoke the subscriber right away on the same thread for a new subscriber.
+     * <p>
+     * This is the preferred way to get a value from a configuration. Instead of {@code setValue(configValue.getOnce())}
+     * use {@code configValue.get((nv,ov)->setValue(nv)) }
+     * This way, every change to the config file will get forwarded to the object.
+     *</p>
+     *
+     * @param s subscriber
+     * @param service service subscribing to dependency topic
+     * @return this topic
+     */
+    public Topic subscribe(Subscriber s, GreengrassService service) {
```

Comment:
why are you adding this? This is highly specific logic for services which should not exist in the config.

Reply:
This is oddly placed in the Topic class, should be in GreengrassService

Reply:
You mean the new subscribe method or the topic watchers?

Reply:
Both

Reply:
Why does this need to be done this way? Why can this code not live in the service class?

Reply:
I wanted to maintain the original definition of `.subscribe` while also allowing `GreengrassService` to store a reference of the dependency watcher